demosplan/DemosPlanCoreBundle/Addon/AddonRegistry.php

```diff
@@ -12,205 +12,104 @@
 
 namespace demosplan\DemosPlanCoreBundle\Addon;
 
-use Carbon\Carbon;
-use Composer\Package\PackageInterface;
-use demosplan\DemosPlanCoreBundle\Addon\Composer\PackageInformation;
+use ArrayAccess;
+use DemosEurope\DemosplanAddon\Permission\PermissionInitializerInterface;
 use demosplan\DemosPlanCoreBundle\Exception\AddonException;
-use demosplan\DemosPlanCoreBundle\Utilities\DemosPlanPath;
-use Exception;
-use Symfony\Component\Yaml\Yaml;
-use Tightenco\Collect\Support\Collection;
+use Iterator;
 
 /**
  * This is the central information repository about all addons installed on this system and their configuration.
  */
-class AddonRegistry
+class AddonRegistry implements Iterator, ArrayAccess
 {
-    public const ADDON_DIRECTORY = '/addons/';
-    public const ADDON_CACHE_DIRECTORY = '/addons/cache/';
+    /** @var array<string, AddonInfo> */
+    private array $addonInfos;
+    private int $iteratorIndex;
 
     /**
-     * Composer package type for addon packages.
+     * @var array<int, string>
      */
-    public const ADDON_COMPOSER_TYPE = 'demosplan-addon';
-
-    private const ADDON_YAML_INLINE_DEPTH = 100;
-
-    private Collection $addons;
-
-    private PackageInformation $installedAddons;
+    private array $iteratorKeys;
 
     public function __construct()
     {
-        $this->installedAddons = new PackageInformation();
-        $this->addons = collect();
-
-        $this->loadAddonInformation();
+        $this->addonInfos = [];
     }
 
-    /**
-     * Reads addon information from the configuration file for all installed addons.
-     */
-    private function loadAddonInformation(): void
+    public function boot(array $addonInfos = [])
     {
-        if (file_exists(DemosPlanPath::getRootPath('addons/addons.yaml'))) {
-            $configFile = Yaml::parseFile(DemosPlanPath::getRootPath('addons/addons.yaml'));
-            if (is_array($configFile) && array_key_exists('addons', $configFile) && is_array($configFile['addons'])) {
-                $this->addons = \collect($configFile['addons']);
-            }
+        if ([] !== $this->addonInfos) {
+            // TODO: throw exception? go into existential crisis? are dinosaurs even extinct?
+            // no they're not, have you heard about chickens?
```

Comment:
I think this should be done and not left as a TODO.

Reply:
Fixed in [72ddb80](https://github.com/demos-europe/demosplan-core/pull/640/commits/72ddb800a9cb35f0f573c39780919e9f96e2c094)